apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,408 @@
+| ESM source code using `import`    | ESM: consumers `import` your package                             | [ESM source and distribution](#esm-source-and-distribution)                                 |
+| ESM: source code uses `import`    | CJS & ESM: consumers either `require()` or `import` your package | [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution) |
+
+### CJS source and distribution
+
+This the "Rum & Coke" of packages: pretty difficult to mess up. Essentially just declare the package’s exports via the [`"exports"`](https://nodejs.org/api/packages.html#conditional-exports) field/field-set.
+
+**Working example**: [cjs-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/cjs-distro)
+
+```json
+{
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" }, // optional, but kind
+  "exports": {
+    ".": "PATH/TO/DIST/CODE/ENTRYPOINT.js", // ex "./dist/index.js"
+    "./package.json": "./package.json" // ensure this file is importable
+  }
+}
+```
+
+Note that `packageJson.exports["."] = filepath` is shorthand for `packageJson.exports["."].default = filepath`
+
+### CJS source and only ESM distribution
+
+The "Gin & Tonic" of packages: This takes a small bit of finesse but is also pretty straight-forward.
+
+**Working example**: [cjs-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/esm-distro)
+
+```json
+{
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" }, // optional, but kind
+  "exports": {
+    ".": "PATH/TO/DIST/CODE/ENTRYPOINT.mjs", // ex "./dist/index.mjs"
+    "./package.json": "./package.json" // ensure this file is importable
+  }
+}
+```
+
+The [`.mjs`](https://nodejs.org/api/esm.html#enabling) file extension is a trump-card: it will override **any** other configuration and the file will be treated as ESM. Using this file extension is necessary because `packageJson.exports.import` does **NOT** signify that the file is ESM (contrary to common, if not universal, misperception), only that it is the file to be used when the package is imported (ESM _can_ import CJS. See [Gotchas](#gotchas) below).
+
+The [`"engines"`](https://nodejs.dev/learn/the-package-json-guide#engines) field provides both a human-friendly and a machine-friendly indication of with which version(s) of Node.js the package is compatible. Depending on the package manager used, an exception may be thrown causing the installation to fail when the consumer is using an incompatible version of Node.js (which can be very helpful to consumers). Including this field here will save a lot of headache for consumers with an older version of Node.js who cannot use the package.
+
+### CJS source and both CJS & ESM distribution
+
+You have a few options:
```

Comment:
this seems to presume you have multiple things to export - is there an example of when you only have one thing to export? (ie, a single-responsibility package)

Reply:
Do they? What makes it seem there are multiple or single exports?